HACKING.adoc

```diff
@@ -84,6 +84,15 @@ You can also query asciidoctor's documentation:
 
     [4] pry(#<Asciidoctor::Section>)> ? find_by
 
+If you install the `pry-byebug` gem you get additional debugging capabilities.
+See the gem's documentation for details.
+
+Since 1.1.0, templates are compiled. It is easier to inject the debug
+triggering statements and use the templates directly instead of debugging
+compiled templates. You can call the slim templates directly with:
+
+    bundle exec asciidoctor --trace -T templates/ examples/customcss.adoc
```

Comment:
You must also load `slim-htag` (`-r slim-htag`).

Reply:
I can't tell you what's going on but it's not necessary. With or without -r slim-htag it will yield the same files:

```
$ bundle exec asciidoctor --trace -T templates/ examples/customcss.adoc
$ sha1sum examples/customcss.html 
6caf3d92da489f44919cefd8d9b803d2aeee559a  examples/customcss.html
$ bundle exec asciidoctor --trace -T templates/ -r slim-htag examples/customcss.adoc
$ sha1sum examples/customcss.html
6caf3d92da489f44919cefd8d9b803d2aeee559a  examples/customcss.html
```

To make sure I'm not doing anything wrong, I made further tests:

```
# calling asciidoctor-revealjs straight (after a fresh rake build)
$ bundle exec asciidoctor-revealjs examples/customcss.adoc 
$ sha1sum examples/customcss.html 
6caf3d92da489f44919cefd8d9b803d2aeee559a  examples/customcss.html

# modifying templates
$ echo -e "p\n  | Intruder alert" >> templates/document.html.slim 
# running without -r slim-htag
$ bundle exec asciidoctor --trace -T templates/ examples/customcss.adoc
$ sha1sum examples/customcss.html 
fdcceb5a1bf892713f9cec52b7902249eaff65d5  examples/customcss.html
# ^ output changed

# w/o a rake build, still get same file as before
$ bundle exec asciidoctor-revealjs examples/customcss.adoc 
$ sha1sum examples/customcss.html 
6caf3d92da489f44919cefd8d9b803d2aeee559a  examples/customcss.html
```

So I would assess that the `-r slim-htag` is not necessary. At least in my environment.